src/fmt.cc

```diff
@@ -0,0 +1,85 @@
+module;
+#ifndef FMT_MODULE_NAME
+#define FMT_MODULE_NAME fmt
+#endif
+// put all implementation-provided headers into the global module fragment
+// to prevent attachment to this module
+#include <cctype>
+#include <cerrno>
+#include <climits>
+#include <cmath>
+#include <cstdarg>
+#include <cstddef>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <ctime>
+#include <cwchar>
+
+#include <algorithm>
+#include <chrono>
+#include <exception>
+#include <functional>
+#include <initializer_list>
+#include <iterator>
+#include <limits>
+#include <locale>
+#include <memory>
+#include <ostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#if _MSC_VER
+#  include <intrin.h>  // _BitScanReverse[64], _BitScanForward[64], _umul128
+#endif
+#ifdef _WIN32
+#  include <io.h>
+#endif
+
+// os.h
+#ifdef FMT_WITH_OS
+  #include <cstdlib>
+  #include <clocale>
+  #if __has_include(<winapifamily.h>)
+    #include <winapifamily.h>
+  #endif
+  #if __has_include(<fcntl.h>)
+    #include <fcntl.h>
+  #endif
+#endif
+
+export module FMT_MODULE_NAME;
+
+#define FMT_MODULE_EXPORT export
+#define FMT_MODULE_EXPORT_BEGIN export {
+#define FMT_MODULE_EXPORT_END }
+
+#define FMT_USE_NONTYPE_TEMPLATE_PARAMETERS 1
+
+// all library-provided declarations and definitions
+// must be in the module purview to be exported
+#include "fmt/format.h"
+#include "fmt/args.h"
+#include "fmt/color.h"
+#include "fmt/compile.h"
+#include "fmt/locale.h"
+#include "fmt/ostream.h"
+#include "fmt/ranges.h"
```

Comment:
These two should be moved to separate modules because in many cases they should be disabled / explicitly opted in.

Reply:
`ostream.h` is included through `printf.h` anyway. 
`ranges.h` can be implemented in a different module if it neither requires any macro from other parts of {fmt} nor any non-exported identifier. Not to mention that teensy tiny modules are not recommended afaik.
What do you think about conditionally including these parts of {fmt} by macro definitions from the command line like it is already proposed for `os.h`? So users of the module can decide on their own if they want this or not.

Reply:
The currently existing dependency levels are as such:

1
core.h

2
args.h -> core.h
format.h -> core.h

3
compile.h -> format.h
ostream.h -> format.h
ranges.h -> format.h
color.h -> format.h
locale.h -> format.h
os.h -> format.h
format-inl.h -> format.h

4
printf.h -> ostream.h
chrono.h -> format.h locale.h

Reply:
Not that I have a say in this, but why split it up and/or have conditional inclusion (this also applies to OS)? In principle, it shouldn't matter how much stuff is included in the module and it's not like fmt has no/few standard library dependencies without them.

Reply:
I'm not sure if I get your point here. You are right about your comment on the amount of stuff included in a module, but why do you mention the standard library? Do you propose not to carve out anything from inclusion into the module interface? If so we are on the same page. So far I've excluded `os.h` for the simple fact that it doesn't matter: nothing is exported from there because there doesn't seem to exist a documented interface that hints on which declarations to export. Additional exports can be added in a subsequent PR by tagging them accordingly. IIUC, Victor's concerns are about template specializations they bring into the TUs that import {fmt}. I have hardly any clue about `ranges.h` in this regard but I cannot really see a problem with `ostream.h`.

Reply:
It must be split because ostream and ranges provide formatter specialization that should not be available by default.

Reply:
Which then rules out `printf.h`, too. Oh well...

Reply:
`printf.h` dependency on `ostream.h` is accidental and will be removed by https://github.com/fmtlib/fmt/commit/b9ab5c8836bbffbe0a877f64d6faef8fbf4fd394.